packages/preferences/src/browser/views/components/preference-node-renderer.ts

```diff
@@ -170,6 +175,65 @@ export abstract class PreferenceLeafNodeRenderer<ValueType extends JSONValue, In
         this.inspection = this.preferenceService.inspect<ValueType>(this.id, this.scopeTracker.currentScope.uri);
     }
 
+    protected buildMarkdownRenderer(): MarkdownRenderer {
+        return new MarkdownRenderer()
+            .modify('a', link => {
+                const href = link.href;
+                link.href = '#';
+                link.title = href;
+                link.addEventListener('click', e => this.openLink(e, href));
+            })
+            .modify('code', code => {
+                const innerText = code.innerText;
+                // Linked preferences always start and end with `#`
+                if (innerText.startsWith('#') && innerText.endsWith('#')) {
+                    const preferenceId = innerText.substring(1, innerText.length - 1);
+                    const preferenceNode = this.model.getNodeFromPreferenceId(preferenceId);
+                    if (preferenceNode) {
+                        let name = this.labelProvider.getName(preferenceNode);
+                        const prefix = this.labelProvider.getPrefix(preferenceNode);
+                        if (prefix) {
+                            name = prefix + name;
+                        }
+                        const link = document.createElement('a');
+                        link.addEventListener('click', e => this.selectPreference(e, preferenceId));
+                        link.innerText = name;
+                        link.title = `#${preferenceId}`;
+                        link.href = '#';
+                        return link;
+                    }
+                }
+                return code;
+            });
+    }
+
+    protected openLink(event: MouseEvent, href: string): void {
+        event.preventDefault();
+        event.stopPropagation();
+        // Exclude right click
+        if (event.which < 3) {
+            window.open(href, '_blank', 'noopener');
+        }
+    }
+
+    protected async selectPreference(event: MouseEvent, preferenceId: string): Promise<void> {
+        event.preventDefault();
+        event.stopPropagation();
+        // Exclude right click
+        if (event.which < 3) {
+            const selector = `li[data-pref-id="${preferenceId}"]`;
+            const element = document.querySelector(selector);
+            if (element) {
+                if (element.clientHeight === 0 || element.clientWidth === 0) {
```

Comment:
There should be a more elegant check for this - we should be able to get the renderer and check whether it's `hidden`, but there isn't, so this works alright. Mea culpa, since I wrote all this :-/.

Reply:
Yeah, I wasn't too happy with the solution as well, since it directly operates on the DOM.

I made the check a bit more sophisticated, using the `hidden` class and only selecting preference nodes which don't appear in the commonly-used group. Are you fine with this approach (at least for this PR)?

Reply:
Checking for the `hidden` class seems safer than checking the size. 👍